### PR TITLE
Feature/architecture fixes

### PR DESCRIPTION
**Pull request by Redjik**
returning tests and fucntionality

### CODE REVIEW

**Comment by zakhenry** [comment]
@Redjik I have added a test for the pagination as the test that you added didn't actually test the pagination length. Can you please revise the implementation so the tests passes (review my added assertion as I may have misunderstood the structure, but I believe the paginator should only return the number of entities from the data).

Also, I am concerned that the custom `LengthAwarePaginator` has performance issues as we are passing the entire collection to be reduced by it, rather than using (Eloquent paginated results)[http://laravel.com/docs/5.1/pagination#paginating-eloquent-results] to only retrieve the needed rows from the database. The performance implications of effectively doing a `SELECT * FROM articles` when there is thousands of articles should be considered.

File: api/tests/TransformerTest.php [hunk 1]
```diff
@@ -96,23 +102,28 @@ public function testRelationSelf()
...
+        $this->assertCount($perPage, $collection['data']);
+    }
+
+    public function testParseIncludes()
+    {
+        $entity = factory(App\Models\TestEntity::class)->make();
+
+        $this->service->parseIncludes('foobar');
+
+        $item = $this->service->item($entity);
```
Comment (zakhenry) [inline api/tests/TransformerTest.php]: This test is dangerous as it will mark code as being covered, but it is not actually being tested.

Comment (Redjik) [inline api/tests/TransformerTest.php]: I've just returned it as it was before =)))
Do we really need parse includes anyway?

**Comment by zakhenry** [comment]
@Redjik can you make sure the pagination works, or if you're going to complete pagination in a separate branch remove that test entirely.

**Pull request merged**